### PR TITLE
C_CPP: Overloading Unary Operators Using 'friend' Functions

### LINKED ISSUE #6547: C_CPP: Overloading Unary Operators Using 'friend' Functions
### Description

Description

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by 124015001**
<hr>

## Description 📜

Explained about how to Overload Unary Operator with 'friend' functions. Also, performed a program as an example explaining the same.
Fixes #6547

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Review by 124015001 — COMMENTED** [review]
corrected typos

**Comment by iam-sakshi** [comment]
can I work on this issue?

**Comment by 124015001** [comment]
@iam-sakshi This is a pull request my friend. You can create an issue and assign it to yourself by writing '/assign' the same way.

**Comment by iam-sakshi** [comment]
okay , sorry and thank you

**Comment by vasu-1** [comment]
> okay , sorry and thank you

you can create your own unique issue !!!

File: C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Overloading Unary Operators Using 'friend' Functions
+
+Here, the operator functions is not the member function of any class. Hence, in this scenario if an operator has to be overloaded to make it work on a class object, the operator function has to be declared as a **friend** to that class.
```
Comment (vasu-1) [inline C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md:3]:
```suggestion
Here, the operator function is not the member function of any class. Hence, in this scenario if an operator has to be overloaded to make it work on a class object, the operator function has to be declared as a **friend** to that class.  
```

File: C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Overloading Unary Operators Using 'friend' Functions
+
+Here, the operator functions is not the member function of any class. Hence, in this scenario if an operator has to be overloaded to make it work on a class object, the operator function has to be declared as a **friend** to that class.  
+
+In other words, when operator function is *friend* to a class, then we can call that function as a "friend operation function."  
+
+The synatx for defining a friend operator function to overload an unary operator is as follows:
```
Comment (vasu-1) [inline C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md:7]:
```suggestion
The syntax for defining a friend operator function to overload a unary operator is as follows:  
```

File: C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+where  
+
+*friend* -> keyword to make a non-member function as a friend to a class  
+*return_type* -> type of the value returned by the friend operator function  
+*operator* -> keyword to create an operator function  
+*opr* -> unary operator that is to be overloaded  
+*argument* -> is the formal argument referring to class object on which the operator overloading is being done.  
+
+Here's a program to overload an unary minus '-' operator using friend functions.  
+
```
Comment (vasu-1) [inline C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md:29]:
```suggestion
*argument* -> is the formal argument referring to the class object on which the operator overloading is being done.  

Here's a program to overload a unary minus '-' operator using friend functions.  

```

File: C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+   //function definition
+}
+```  
+where  
+
+*friend* -> keyword to make a non-member function as a friend to a class  
+*return_type* -> type of the value returned by the friend operator function  
+*operator* -> keyword to create an operator function  
+*opr* -> unary operator that is to be overloaded  
+*argument* -> is the formal argument referring to class object on which the operator overloading is being done.
```
Comment (vasu-1) [inline C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md:26]:
```suggestion
- *friend* -> keyword to make a non-member function as a friend to a class  
- *return_type* -> type of the value returned by the friend operator function  
- *operator* -> keyword to create an operator function  
- *opr* -> unary operator that is to be overloaded  
- *argument* -> is the formal argument referring to class object on which the operator overloading is being done.  
```

do some markdown formatting like this.

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Changes Required

**Review by 124015001 — COMMENTED** [review]
updated suggestions

**Comment by 124015001** [comment]
@vasu-1

**Review by vasu-1 — APPROVED** [review]
@Saviour1001

**Pull request merged**